### PR TITLE
better oauth2 ui integration

### PR DESCRIPTION
**Pull request by chicco785**
1. single sign out
1. better ui in case of only oauth provider used

fix issue: https://github.com/gravitee-io/issues/issues/1060

### CODE REVIEW

**Comment by brasseld** [comment]
Hi @chicco785 

Thanks for this PR.

Don't you think it would be better to create two separate issues and PRs ?
One for login-mode (form, oauth, both)
One for single sign out

File: constants.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
   "managementTitle": "Gravitee.io Management",
   "devMode": false,
   "userCreationEnabled": true,
+  "localLoginEnabled": true,
```
Comment (brasseld) [inline constants.json]: I'd prefer to put this property in `constants.tpl.json` file.
Also, I think it's better to set this property as part of the `authentication` section (https://github.com/gravitee-io/gravitee-management-webui/blob/master/constants.tpl.json#L26).

File: src/portal/components/navbar/navbar.component.ts [hunk 2]
```diff
@@ -124,5 +128,18 @@ export const NavbarComponent: ng.IComponentOptions = {
...
+      AuthenticationService.authenticate(provider)
+        .then( () => {
+          UserService.current().then( () => {
+            vm.$rootScope.$broadcast('graviteeUserRefresh');
+          });
+        })
+        .catch( () => {});
+    };
+
+    vm.isOnlyOAuth = !Constants.localLoginEnabled && vm.providers.length == 1;
```
Comment (brasseld) [inline src/portal/components/navbar/navbar.component.ts]: Does it means that if social authentication is enabled, it is not under the `oauth` scope ?

File: src/user/login/login.html [hunk 3]
```diff
@@ -23,7 +23,7 @@
           <label>Log In</label>
         </div>
 
-        <form name="formUser" ng-submit="$ctrl.login($event)" layout="column">
+        <form name="formUser" ng-submit="$ctrl.login($event)" layout="column" ng-if="$ctrl.localLoginEnabled">
```
Comment (brasseld) [inline src/user/login/login.html]: What is the reason to show the login view if `localLogin` is disabled ?
IMHO, login view is more-or-less useless if we are doing only oauth authentication.

WDYT ?

File: constants.json (hunk 1 above)
Comment (chicco785) [reply constants.json]: will do

File: src/portal/components/navbar/navbar.component.ts (hunk 2 above)
Comment (chicco785) [reply src/portal/components/navbar/navbar.component.ts]: it is only a way to compute the fact that local login is disable and that there is only one single identity provider (it does not really matter if generic oauth2 or google ect). 
in this case we set the link to the external provider on the top right, if local login is enabled or there are more than one external identity provider, we open the form where the user can choose between the list of external identity providers.

File: src/user/login/login.html (hunk 3 above)
Comment (chicco785) [reply src/user/login/login.html]: (see above reply) this is to handle the case where there are multiple external idp. let's say google and a custom oauth. having all of them in the top bar will not look too good, so we display the login form with the list of the providers.

please let me know if you prefer a different logic.

File: constants.json (hunk 1 above)
Comment (chicco785) [reply constants.json]: done

File: src/user/login/login.controller.ts [hunk 4]
```diff
@@ -37,7 +37,7 @@ class LoginController {
   ) {
     'ngInject';
     this.userCreationEnabled = Constants.userCreationEnabled;
-    this.localLoginEnabled = Constants.localLoginEnabled;
+    this.localLoginEnabled = Constants.authentication.localLoginEnabled;
```
Comment (brasseld) [inline src/user/login/login.controller.ts]: What if `Constants.authentication` is undefined  (no authentication config in constants.json) ?

**Comment by brasseld** [comment]
Please squash two commits into a single one to keep a clean commit history

File: src/user/login/login.controller.ts (hunk 4 above)
Comment (chicco785) [reply src/user/login/login.controller.ts]: fixed

**Comment by chicco785** [comment]
commit history is now clean

**Comment by brasseld** [comment]
Hi @chicco785 ,

You miss to include a reference to the issue in your commit message.
It should be:

```
feat(oauth2): better oauth2 ui integration

1. single sign out
1. better ui in case of only oauth provider used

Closes gravitee-io/issues#1060
```

**Comment by chicco785** [comment]
ok sorry :) thx for accepting my contrib!

**Comment by chicco785** [comment]
@brasseld when do you think the PR will be landed? Just to know if we should use an official release or a custom one to have it.

Thx :)

**Comment by brasseld** [comment]
It will be accepted for the next 1.14.x scheduled for 27th February.

I'm just waiting for an issue to be accepted (user anonymization) because it has a lot some impacts on user management.

**Comment by brasseld** [comment]
Nice job @chicco785 👍 

Thanks a lot !

**Pull request merged**